pkg/argparser/args.go

```diff
@@ -112,6 +112,13 @@ func GetArgs(args string, state *state.HelmState) []string {
 		}
 	}
 
+	if len(state.HelmDefaults.PostRenderer) > 0 {
+		argArr = append(argArr, fmt.Sprintf("--post-renderer=\"%s\"", state.HelmDefaults.PostRenderer))
+	}
+	for _, arg := range state.HelmDefaults.PostRendererArgs {
+		argArr = append(argArr, fmt.Sprintf("--post-renderer-args=\"%s\"", arg))
+	}
+
```

Comment:
I believe this is not the right place to add this code- The helmDefaults args, the under pinning of the `--args` flag of helmfile, has known to be not well-designed, and being removed in Helmfile v1. The main difficulty is that Helmfile has no way to know which args specified in the helmDefaults.args to be passed to which helm sub-commands (install, uninstall, list, diff, etc) a helmfile command is going to run.

More importantly, the post renderer should be passed down to `chartify` when you're going to use a directory of raw k8s manifests as a chart, or a kustomize config as a chart, or you're using some chartify-provided features like ad-hoc dependencies and ad-hoc patching(jsonPatches and stragicMergePatches, forceNamespace, and so on).

Still curious- did this actually work on your machine? 🤔

Reply:
well,   the `--args` flag of helmfile is easy to use indeed, such as:
```
helmfile -e ksp -f ti-k8s/helmfile.yaml -l chart=mysql  --args="--post-renderer=`pwd`rewrite-ksp.sh"
```
as the `post renderer` accept a abs path of exec script, this works well in our project; 

though it is more `chartify` to special each flags to parse the flags to helm exactly； but based on what I know， it is need to refactor lots of  code,  may be change the `helmexec.Interface` and add some set methods
？  and  invoke at every  place of  `helmexec.SetExtraArgs` ?

So what is the good way to implement this feature？  may give some Guidelines please?   

thanks.

Reply:
> this works well in our project;

Yep. But it's not guaranteed to work. We don't have a complete design doc and tests to make sure your usage of "--args" works. My understanding was that Helmfile must have been implemented to pass the args to every helm sub-commands, regardless of if the sub-command supports the set of args or not.

Reply:
> the good way to implement this feature

I'd recommend following the same pattern that is used for other Helmfile flags and fields. Propagate Helmfile flags to helmDefaults, inherit helmDefaults to releases[] fields if necessary, and finally compose chartify parameters and helm flags from the final releases[] entry fields.

Reply:
@tanguofu Can you implement the change suggested by @mumoshu? The solution with the possibility to set the post-rendering through flags, helmDefaults or per release sounds perfect. Missing Post-rendering is the only thing stopping me from switching to Helmfile, so I'd be very grateful. Keep up the good work, guys!

Reply:
> ok, i will  implement the change, please just wait a moment.